include/pybind11/detail/common.h

```diff
@@ -447,23 +446,33 @@
 PYBIND11_WARNING_PUSH
 PYBIND11_WARNING_DISABLE_CLANG("-Wgnu-zero-variadic-macro-arguments")
 #define PYBIND11_MODULE(name, variable, ...)                                                      \
-    static ::pybind11::module_::module_def PYBIND11_CONCAT(pybind11_module_def_, name)            \
-        PYBIND11_MAYBE_UNUSED;                                                                    \
-    PYBIND11_MAYBE_UNUSED                                                                         \
+    static ::pybind11::module_::module_def PYBIND11_CONCAT(pybind11_module_def_, name);           \
+    static PyModuleDef_Slot PYBIND11_CONCAT(pybind11_module_slots_, name)[5];                     \
+    static int PYBIND11_CONCAT(pybind11_exec_, name)(PyObject *);                                 \
     static void PYBIND11_CONCAT(pybind11_init_, name)(::pybind11::module_ &);                     \
     PYBIND11_PLUGIN_IMPL(name) {                                                                  \
         PYBIND11_CHECK_PYTHON_VERSION                                                             \
         PYBIND11_ENSURE_INTERNALS_READY                                                           \
-        auto m = ::pybind11::module_::create_extension_module(                                    \
-            PYBIND11_TOSTRING(name),                                                              \
-            nullptr,                                                                              \
-            &PYBIND11_CONCAT(pybind11_module_def_, name),                                         \
-            ##__VA_ARGS__);                                                                       \
+        PYBIND11_CONCAT(pybind11_module_slots_, name)[0].slot = Py_mod_exec;                      \
```

Comment:
We have `PYBIND11_CONCAT(pybind11_module_slots_, name)` here 4 times.

IIUC, we're inside a `{ ... }` scope.

Could something like this work:
```
    PyModuleDef_Slot *slots = PYBIND11_CONCAT(pybind11_module_slots_, name);
```
So that we can just refer to `slots` below?

Reply:
Done